src/main.cpp

```diff
@@ -893,13 +900,16 @@ static bool run_dedicated_server(const GameParams &game_params, const Settings &
 
 static bool migrate_map_database(const GameParams &game_params, const Settings &cmd_args)
 {
-	std::string migrate_to = cmd_args.get("migrate");
 	Settings world_mt;
+	set_default_settings(&world_mt);
+	override_default_settings(&world_mt, g_main_settings);
 	std::string world_mt_path = game_params.world_path + DIR_DELIM + "world.mt";
+
 	if (!world_mt.readConfigFile(world_mt_path.c_str())) {
 		errorstream << "Cannot read world.mt!" << std::endl;
 		return false;
 	}
+	g_settings = &world_mt;
```

Comment:
Pointer assigned to address of local variable. If you want to keep this, please ensure `g_setting` points to valid data when the function returns.

Reply:
:O